bats/tests/helpers/profile.bash

```diff
@@ -0,0 +1,348 @@
+case $OS in
+darwin)
+    PROFILE_SYSTEM_DEFAULTS=/Library/Preferences/io.rancherdesktop.profile.defaults.plist
+    PROFILE_SYSTEM_LOCKED=/Library/Preferences/io.rancherdesktop.profile.locked.plist
+    PROFILE_USER_DEFAULTS="${HOME}${PROFILE_SYSTEM_DEFAULTS}"
+    PROFILE_USER_LOCKED="${HOME}${PROFILE_SYSTEM_LOCKED}"
+    ;;
+linux)
+    PROFILE_SYSTEM_DEFAULTS=/etc/rancher-desktop/defaults.json
+    PROFILE_SYSTEM_LOCKED=/etc/rancher-desktop/locked.json
+    PROFILE_USER_DEFAULTS="${HOME}/.config/rancher-desktop.defaults.json"
+    PROFILE_USER_LOCKED="${HOME}/.config/rancher-desktop.locked.json"
+    ;;
+windows)
+    PROFILE='Software\Policies\Rancher Desktop'
+    PROFILE_SYSTEM_DEFAULTS="HKLM\\${PROFILE}\\Defaults"
+    PROFILE_SYSTEM_LOCKED="HKLM\\${PROFILE}\\Locked"
+    PROFILE_USER_DEFAULTS="HKCU\\${PROFILE}\\Defaults"
+    PROFILE_USER_LOCKED="HKCU\\${PROFILE}\\Locked"
+
+    # The legacy profiles (for both system and user) are supported for backward
+    # compatibility with Rancher Desktop 1.8.x. For BATS purposes the legacy
+    # user profiles have the advantage of being writable without admin rights.
+    PROFILE='Software\Rancher Desktop\Profile'
+    PROFILE_SYSTEM_LEGACY_DEFAULTS="HKLM\\${PROFILE}\\Defaults"
+    PROFILE_SYSTEM_LEGACY_LOCKED="HKLM\\${PROFILE}\\Locked"
+    PROFILE_USER_LEGACY_DEFAULTS="HKCU\\${PROFILE}\\Defaults"
+    PROFILE_USER_LEGACY_LOCKED="HKCU\\${PROFILE}\\Locked"
+    ;;
+esac
+
+PROFILE_SYSTEM=system
+PROFILE_SYSTEM_LEGACY=system-legacy
+PROFILE_USER=user
+PROFILE_USER_LEGACY=user-legacy
+
+PROFILE_DEFAULTS=defaults
+PROFILE_LOCKED=locked
+
+# Default location is a writable user location
+if is_windows; then
+    PROFILE_LOCATION=$PROFILE_USER_LEGACY
+else
+    PROFILE_LOCATION=$PROFILE_USER
+fi
+PROFILE_TYPE=$PROFILE_DEFAULTS
+
+# profile_location is a registry key on Windows, or a filename on macOS and Linux.
+profile_location() {
+    local profile=$(to_upper "profile_${PROFILE_LOCATION}_${PROFILE_TYPE}" | tr - _)
+    echo "${!profile}"
+}
+
+# Execute command for each profile
+foreach_profile() {
+    local locations=("$PROFILE_SYSTEM" "$PROFILE_USER")
+    if is_windows; then
+        locations+=("$PROFILE_SYSTEM_LEGACY" "$PROFILE_USER_LEGACY")
+    fi
+
+    local PROFILE_LOCATION PROFILE_TYPE
```

Comment:
Because these are the names we ultimately need to set before invoking the command. I originally passed them like this:

```
PROFILE_LOCATION=$location PROFILE_TYPE=$type "$@"
```

because I only want to set them for the call, but restore them to their original values when `foreach_profile` returns. Then I realized that `local` would do that for me automatically, so I can just set them here without the additional variables.